### PR TITLE
sb3: set relationships as lazy load.

### PR DESCRIPTION
**Pull request by mshima**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/server/templates/entity/src/main/java/package/domain/Entity.java.jhi.jakarta_persistence.ejs [hunk 1]
```diff
@@ -77,14 +77,14 @@ import org.hibernate.type.SqlTypes;
 <%_ for (const relationship of relationships) { -%>
 <&_ if (fragment.relationship<%- relationship.relationshipNameCapitalized %>AnnotationSection) { -&>
   <%_ if (relationship.relationshipOneToMany) { -%>
-    @OneToMany(mappedBy = "<%= relationship.otherEntityRelationshipName %>")
+    @OneToMany(fetch = FetchType.LAZY, mappedBy = "<%= relationship.otherEntityRelationshipName %>")
```
Comment (Tcharl) [inline generators/server/templates/entity/src/main/java/package/domain/Entity.java.jhi.jakarta_persistence.ejs:80]: Arent't they lazy by default?

File: generators/server/templates/entity/src/main/java/package/domain/Entity.java.jhi.jakarta_persistence.ejs [hunk 2]
```diff
@@ -110,7 +110,7 @@ import org.hibernate.type.SqlTypes;
     }, allowSetters = true)
       <%_ } _%>
       <%_ if (relationship.ownerSide) { -%>
-    @OneToOne<% if (relationship.relationshipRequired) { %>(optional = false)<% } %>
+    @OneToOne(fetch = FetchType.LAZY<% if (relationship.relationshipRequired) { %>, optional = false<% } %>)
```
Comment (Tcharl) [inline generators/server/templates/entity/src/main/java/package/domain/Entity.java.jhi.jakarta_persistence.ejs:113]: This one is usually eager, but for the goodwill ^^

Comment (mshima) [reply generators/server/templates/entity/src/main/java/package/domain/Entity.java.jhi.jakarta_persistence.ejs:113]: I also thought think (see https://github.com/jhipster/generator-jhipster/pull/19791#issuecomment-1312003467), but this means other others relationships will load those relationships too.
IMO the safest right now it to make everything lazy.

File: generators/server/templates/entity/src/main/java/package/domain/Entity.java.jhi.jakarta_persistence.ejs (hunk 1 above)
Comment (mshima) [reply generators/server/templates/entity/src/main/java/package/domain/Entity.java.jhi.jakarta_persistence.ejs:80]: @Tcharl I also thought, but see https://github.com/jhipster/generator-jhipster/pull/19791#issuecomment-1311988418 and related discussion.

**Review by mraible — APPROVED** [review]

File: generators/server/templates/entity/src/main/java/package/domain/Entity.java.jhi.jakarta_persistence.ejs (hunk 1 above)
Comment (Tcharl) [reply generators/server/templates/entity/src/main/java/package/domain/Entity.java.jhi.jakarta_persistence.ejs:80]: Weird... Maybe mentionning to remove the statement after hibernate 6 could be a plus

**Pull request merged**